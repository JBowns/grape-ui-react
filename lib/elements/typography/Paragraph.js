'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Paragraph = undefined;

var _templateObject = _taggedTemplateLiteral(['\n    ', '\n    margin: 0 0 1rem;\n  '], ['\n    ', '\n    margin: 0 0 1rem;\n  ']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _textStyles = require('./textStyles');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Paragraph = function Paragraph(props) {
  var otherProps = _objectWithoutProperties(props, []);

  var actualBase = (0, _textStyles.textStylesBase)(props);
  var ProtoParagraph = _styledComponents2.default.p(_templateObject, actualBase);
  return _react2.default.createElement(ProtoParagraph, otherProps);
};
exports.Paragraph = Paragraph;