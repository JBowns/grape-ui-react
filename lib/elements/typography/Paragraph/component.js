"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ParagraphComponent = undefined;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledHelpers = require("../../../utils/styledHelpers");

var _componentHelpers = require("../../../utils/componentHelpers");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var propsToTrim = _objectSpread({
  lead: ''
}, _styledHelpers.spaceProps.propTypes, {}, _styledHelpers.typography.propTypes);

var ParagraphComponent = exports.ParagraphComponent = function ParagraphComponent(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ["children"]);

  return _react2["default"].createElement("p", (0, _componentHelpers.removeSomeProps)(props, Object.keys(propsToTrim)), children);
};

ParagraphComponent.propTypes = {
  children: _propTypes2["default"].any.isRequired,
  lead: _propTypes2["default"].bool
};
ParagraphComponent.defaultProps = {
  lead: false
};