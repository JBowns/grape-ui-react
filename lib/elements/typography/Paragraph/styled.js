"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Paragraph = undefined;

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require("styled-components");

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _styledSystem = require("styled-system");

var _globalStyles = require("../../../global-styles");

var _styledHelpers = require("../../../utils/styledHelpers");

var _component = require("./component");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var _getGlobalStyles = (0, _globalStyles.getGlobalStyles)(),
    gridSchema = _getGlobalStyles.grid;

var fontSizeParagraph = function fontSizeParagraph(props) {
  var lead = props.lead;
  return lead ? (0, _styledHelpers.fontSizeCore)(_objectSpread({}, props, {
    lg: true
  })) : (0, _styledHelpers.fontSizeCore)(props);
};

var fontWeightParagraph = function fontWeightParagraph(props) {
  var lead = props.lead;
  return lead ? (0, _styledSystem.fontWeight)(_objectSpread({}, props, {
    fontWeight: '300'
  })) : (0, _styledSystem.fontWeight)(props);
};

var Paragraph = (0, _styledComponents2["default"])(_component.ParagraphComponent)(_templateObject(), _styledHelpers.colorCore, _styledHelpers.ellipsisCore, _styledHelpers.fontFamilyCore, fontSizeParagraph, fontWeightParagraph, _styledHelpers.letterSpacingCore, _styledHelpers.lineHeightCore, _styledHelpers.fontStyleCore, _styledHelpers.textAlignCore, _styledHelpers.textDecorationCore, _styledSystem.space);
Paragraph.propTypes = _objectSpread({}, _styledHelpers.typography.propTypes, {
  lead: _propTypes2["default"].bool
});
Paragraph.defaultProps = _objectSpread({}, _styledHelpers.defaultStylesBase, {
  lead: false,
  mb: gridSchema.gutter,
  mt: 0
});
/** @component */

exports.Paragraph = Paragraph;