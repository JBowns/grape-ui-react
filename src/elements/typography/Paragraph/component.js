import React from 'react';
import PropTypes from 'prop-types';
import { spaceProps, typography } from 'src/utils/styledHelpers';
import { removeSomeProps } from 'src/utils/componentHelpers';

const propsToTrim = {
  lead: '',
  ...spaceProps.propTypes,
  ...typography.propTypes,
};
export const ParagraphComponent = ({ children, ...props }) => (
  <p {...removeSomeProps(props, Object.keys(propsToTrim))}>
    {children}
  </p>
);
ParagraphComponent.propTypes = {
  children: PropTypes.any.isRequired,
  lead: PropTypes.bool,
};

ParagraphComponent.defaultProps = {
  lead: false,
};
