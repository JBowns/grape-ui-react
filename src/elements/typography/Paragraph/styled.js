import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getGlobalStyles } from 'src/global-styles';
import {
  getFontFamily,
  getFontSize,
  getFontWeight,
  getLetterSpacing,
  getLineHeight,
  getFontStyle,
  getColor,
  getTextAlign,
  getTextDecoration,
  typography,
} from '../textStyles';
import { ParagraphComponent } from './component';

const { grid: gridSchema } = getGlobalStyles();

const getParagraphFontSize = props => {
  const { lead } = props;
  return lead ? getFontSize({ ...props, lg: true }) : getFontSize(props);
};
const getParagraphFontWeight = props => {
  const { lead } = props;
  return lead ? getFontWeight({ ...props, fontWeight: '300' }) : getFontWeight(props);
};

const Paragraph = styled(ParagraphComponent)`
  ${getFontFamily}
  ${getParagraphFontSize}
  ${getParagraphFontWeight}
  ${getLetterSpacing}
  ${getLineHeight}
  ${getFontStyle}
  ${getColor}
  ${getTextAlign}
  ${getTextDecoration}
  margin: 0 0 ${gridSchema.gutter};
  `;

Paragraph.propTypes = {
  ...typography.propTypes,
  lead: PropTypes.bool,
};

Paragraph.defaultProps = {
  lead: false,
};
export { Paragraph };
