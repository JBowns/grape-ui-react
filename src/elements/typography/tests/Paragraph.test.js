import React from 'react';
import { Paragraph } from '../Paragraph';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Paragraph Component', () => {
  it('should return a Paragraph object with font-style: base', () => {
    const component = renderer.create(<Paragraph>Lorem Ipsum</Paragraph>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with font-style: italic', () => {
    const component = renderer.create(<Paragraph italic>Lorem Ipsum</Paragraph>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with font-size: lg', () => {
    const component = renderer.create(<Paragraph lg>Lorem Ipsum</Paragraph>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with font-size: sm', () => {
    const component = renderer.create(<Paragraph sm>Lorem Ipsum</Paragraph>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with text-align: right', () => {
    const component = renderer.create(<Paragraph textAlign="right">Lorem Ipsum</Paragraph>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with font-weight: bold', () => {
    const component = renderer.create(<Paragraph fontWeight="bold">Lorem Ipsum</Paragraph>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
