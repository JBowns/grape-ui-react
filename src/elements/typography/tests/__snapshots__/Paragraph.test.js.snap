// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Component should return a Paragraph object with a lead paragraph stlying 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1.2);
  font-weight: 300;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component should return a Paragraph object with base styling 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component should return a Paragraph object with font-size: lg 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1.2);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component should return a Paragraph object with font-size: sm 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * .8);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component should return a Paragraph object with font-style: italic 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  font-style: italic;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component should return a Paragraph object with font-weight: bold 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: bold;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<p
  className="c0"
  fontWeight="bold"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component should return a Paragraph object with text-align: right 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: right;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;
